server/src/main/java/org/opensearch/transport/CompressibleBytesOutputStream.java

```diff
@@ -68,7 +68,7 @@ final class CompressibleBytesOutputStream extends StreamOutput {
         this.bytesStreamOutput = bytesStreamOutput;
         this.shouldCompress = shouldCompress;
         if (shouldCompress) {
-            this.stream = CompressorFactory.COMPRESSOR.threadLocalOutputStream(Streams.flushOnCloseStream(bytesStreamOutput));
+            this.stream = CompressorFactory.DEFLATE_COMPRESSOR.threadLocalOutputStream(Streams.flushOnCloseStream(bytesStreamOutput));
```

Comment:
The `CompressibleBytesOutputStream` probably has to be changed to use `Compressor` instead of `shouldCompress` flag, it would also help us to use / change the compression on transport level